tests/rustdoc/inline_cross/const-effect-param.rs

```diff
@@ -0,0 +1,29 @@
+// Regression test for issue #116629.
+// Check that we don't render host effect parameters & arguments.
+
+// aux-crate:const_effect_param=const-effect-param.rs
+// edition: 2021
+#![crate_name = "user"]
+
+// Don't render the host param on `load` and the host arg `host` passed to `Resource`.
+// @has user/fn.load.html
+// @has - '//pre[@class="rust item-decl"]' "pub const fn load<R>() -> i32\
+//     where \
+//         R: Resource"
+pub use const_effect_param::load;
+
+// Don't render the host arg `true` passed to `Resource`.
```

Comment:
This one was fixed by my “elide cross-crate default generic args” PR which ofc predates oli's PR, hence no test yet. Feature `effects` synthesizes the param `#[rustc_host_param] const host: bool = true` for `const fn` etc, therefore `true` gets elided since that PR of mine.

This PR introduces a more principled approach and not only elides `host` and `true` but arbitrary const values like `false`. I can't add for `false` yet since `effects` doesn't synthesize such code yet. My "generic const items: support effects" PR (can't link, on mobile rn) would change that and rustdoc would handle it correctly (I've double-checked that).